include/mpi_cpp_tools/infix_iterator.h

```diff
@@ -0,0 +1,51 @@
+// infix_iterator.h
+#if !defined(INFIX_ITERATOR_H_)
+#define  INFIX_ITERATOR_H_
+#include <ostream>
+#include <iterator>
+#include <string>
+
+template <class T, class charT=char, class traits=std::char_traits<charT> >
```

Comment:
Please add a doxygen comment that briefly describes what this is doing.

Also note that content on StackExchange is license with the CC BY-SA 4.0 license.  This means we have to
- give attribution (link to the source, name of the author, and a comment in case you modified anything).
- keep the license (it seems to be a copy-left license). For this, simply add a comment at the top of the file mentioning the license and the link to it: https://creativecommons.org/licenses/by-sa/4.0/ (see also that link for details about the license).